modules/azure/resourceid.go

```diff
@@ -0,0 +1,36 @@
+package azure
+
+import (
+	"errors"
+	"strings"
+)
+
+// GetNameFromResourceID gets the Name from an Azure Resource ID
+func GetNameFromResourceID(resourceID string) string {
+	lastValue, err := GetSliceLastValueE(resourceID, "/")
+	if err != nil {
+		return ""
+	}
+	return lastValue
+}
+
+// GetSliceLastValueE will take a source string and returns the last value when split by the seperaror char
+func GetSliceLastValueE(source string, seperator string) (string, error) {
```

Comment:
This doesn't seem specific to Azure. Perhaps move to `collections` package?

Reply:
Also, this should have dedicated unit tests.

Reply:
Ah, I see some tests below. Those should be moved to `modules/azure/resourceid_test.go`.

Reply:
Moved string/separator/slice/select functionality to collections, not sure what to name the file though.

Reply:
Please see: modules/collections/stringslicevalue.go